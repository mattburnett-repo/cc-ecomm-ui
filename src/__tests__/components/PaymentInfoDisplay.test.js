// 20211217: Wireframe https://wireframe.cc/nx8Dv9

// import { render, screen, fireEvent } from '@testing-library/react';
// import renderer from 'react-test-renderer';
// import userEvent from '@testing-library/user-event';

// import PaymentInfoDisplay from '../../components/payments/PaymentInfoDisplay';

describe('<PaymentInfo /> component tests', () => {
    test.todo('Adding StyledComponents results in \'TypeError: Cannot read properties of undefined (reading \'light\')\' errors. Figure this out someday.')

    // beforeEach(() => render(<PaymentInfoDisplay  />));

    // test.todo('should render')
    // test.todo('should render cart / order summary')
    // test.todo('should render shipping summary')
    // test.todo('should render Stripe Elements')
    // test.todo('should render a snapshot')

    // it('should render', () => {
    //     screen.getByRole('presentation', { name: /payment-info/i})
    // })
    // it('should render cart / order summary', () => {
    //     screen.getByRole('presentation', { name: /cart-summary-info/i} )
    // })
    // it('should render shipping summary', () => {
    //     screen.getByRole('presentation', { name: /shipping-summary-info/i} )
    // })

    // Payment data capture is performed by Stripe elements

    // it('should render a payment types selector', () => {
    //     screen.getByRole('presentation', { name: /^payment-types$/i })
    // })
    // it('should select a payment type', () => {
    //     fireEvent.change(screen.getByRole('presentation', {name: /payment-types-selector/i}), { target: { value: 2 } })
    //     let options = screen.getByRole('presentation', {name: /payment-types-selector/i});
    //     expect(options[0].selected).toBeFalsy();
    //     expect(options[1].selected).toBeTruthy();
    //     expect(options[2].selected).toBeFalsy();
    //     expect(options[3].selected).toBeFalsy();
    //     expect(options[4].selected).toBeFalsy();
    // });

    // it('should display and accept a name on card', () => {
    //     const theElement = screen.getByRole('textbox', {name: /name-on-card/i});
    //     userEvent.type(theElement, 'name-on-card');
        
    //     expect(theElement).toHaveValue('name-on-card');
    // });
    // it('should display and accept a card number', () => {
    //     const theElement = screen.getByRole('textbox', {name: /card-number/i});
    //     userEvent.type(theElement, '1234 1234 1234 1234');
        
    //     expect(theElement).toHaveValue('1234 1234 1234 1234');
    // });
    // it('should display and accept an expiration date', () => {
    //     const theElement = screen.getByRole('textbox', {name: /expiration-date/i});
    //     userEvent.type(theElement, '01/2022');
        
    //     expect(theElement).toHaveValue('01/2022');
    // });
    // it('should display and accept a cvc number', () => {
    //     const theElement = screen.getByRole('textbox', {name: /cvc-number/i});
    //     userEvent.type(theElement, '1234');
        
    //     expect(theElement).toHaveValue('1234');
    // });
    // it('should display and accept a postal code', () => {
    //     const theElement = screen.getByRole('textbox', {name: /postal-code/i});
    //     userEvent.type(theElement, 'BR 549');
        
    //     expect(theElement).toHaveValue('BR 549');
    // });

    // it('should display and click a save payment info button', () => {
    //     let theVal = screen.getByRole('button', {name: /save-payment-info/i});
    //     fireEvent.submit(theVal);
    // })
    // it('should display and click a finish order button', () => {
    //     let theVal = screen.getByRole('button', {name: /finish-order/i});
    //     fireEvent.submit(theVal);
    // });

    // it('creates a snapshot', () => {
    //     const tree = renderer.create(<PaymentInfoDisplay />).toJSON();
    //     expect(tree).toMatchSnapshot();  
    // });
}); // end component
