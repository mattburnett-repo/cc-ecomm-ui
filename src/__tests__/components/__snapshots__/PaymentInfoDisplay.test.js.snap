// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentInfo /> component tests creates a snapshot 1`] = `
<div
  aria-label="payment-info"
  role="presentation"
>
  <h3>
    Payment Info
  </h3>
  <div
    aria-label="cart-summary-info"
    role="presentation"
  >
    Cart / Order has 
    0
     items, total price: FIXME
  </div>
  <div
    aria-label="shipping-summary-info"
    role="presentation"
  >
    Order will ship to 
     
    ,
     
    ,
     
    ,
     
  </div>
  <form
    method="post"
    onSubmit={[MockFunction]}
  >
    <div
      aria-label="payment-types"
      role="presentation"
    >
      <label
        htmlFor="payment-types-selector"
      >
        Payment Types:
      </label>
      <select
        aria-label="payment-types-selector"
        defaultValue="0"
        id="payment-types-selector"
        name="paymentTypes"
        role="presentation"
      >
        <option
          value={1}
        >
          PayPal
        </option>
        <option
          value={2}
        >
          MasterCard
        </option>
        <option
          value={3}
        >
          Visa
        </option>
        <option
          value={4}
        >
          Amex
        </option>
        <option
          value={5}
        >
          EC Karte
        </option>
      </select>
    </div>
    <div>
      <label
        htmlFor="name-on-card"
      >
        Name on Card:
      </label>
      <input
        aria-label="name-on-card"
        id="name-on-card"
        name="nameOnCard"
      />
    </div>
    <div>
      <label
        htmlFor="card-number"
      >
        Card Number:
      </label>
      <input
        aria-label="card-number"
        id="card-number"
        name="cardNumber"
      />
    </div>
    <div>
      <label
        htmlFor="expiration-date"
      >
        Exp. Date:
      </label>
      <input
        aria-label="expiration-date"
        id="expiration-date"
        name="expDate"
      />
    </div>
    <button
      aria-label="finish-order"
      type="submit"
    >
      Finish Order
    </button>
  </form>
</div>
`;
